### PR TITLE
PowerVS: Add DNS record check in base-domain

### PR DESCRIPTION
**Pull request by dharaneeshvrd**
**What this PR does / why we need it**:
This PR trying to add a check during infra creation that If DNS record with the cluster name already exist in base domain, do not allow infra creation.

RH Issue: https://issues.redhat.com/browse/MULTIARCH-2634

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: cmd/infra/powervs/create.go [hunk 1]
```diff
@@ -158,13 +160,15 @@ func NewCreateCommand() *cobra.Command {
 	}
 
 	opts := CreateInfraOptions{
+		Name:          "example",
```
Comment (alvaroaleman) [inline cmd/infra/powervs/create.go:163]: The cluster name is already a config option in provider-agnostic create cluster layer, we shouldn't redefine it her: https://github.com/openshift/hypershift/blob/753942e8319b4a6f7e259591aef4e53defbdc854/cmd/cluster/cluster.go#L44

File: cmd/infra/powervs/create.go [hunk 2]
```diff
@@ -412,6 +416,28 @@ func getCISDomainDetails(baseDomain string) (string, string, error) {
 	return CISCRN, CISDomainID, nil
 }
 
+// checkForExistingDNSRecord check for existing DNS record with the cluster name
+func checkForExistingDNSRecord(options *CreateInfraOptions, CISCRN string, CISDomainID string) error {
+	dnsRecordsV1, err := dnsrecordsv1.NewDnsRecordsV1(&dnsrecordsv1.DnsRecordsV1Options{Crn: &CISCRN, ZoneIdentifier: &CISDomainID, Authenticator: getIAMAuth()})
+	if err != nil {
+		return fmt.Errorf("error creating dns record service %w", err)
```
Comment (alvaroaleman) [inline cmd/infra/powervs/create.go:423]:
```suggestion
		return fmt.Errorf("error creating dns record client: %w", err)
```

Comment (dharaneeshvrd) [reply cmd/infra/powervs/create.go]: Modified.

File: cmd/infra/powervs/create.go (hunk 1 above)
Comment (dharaneeshvrd) [reply cmd/infra/powervs/create.go:163]: But this is on infra command, name defined in cluster command won't be accessible here. It is similar to name defined here for AWS infra https://github.com/openshift/hypershift/blob/main/cmd/infra/aws/create.go#L91

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**